geode-core/src/main/java/org/apache/geode/cache/client/internal/ClientMetadataService.java

```diff
@@ -211,9 +214,12 @@ private ServerLocation getServerLocation(Region region, Operation operation, int
 
     HashMap<ServerLocation, HashSet> serverToKeysMap = new HashMap<ServerLocation, HashSet>();
     HashMap<ServerLocation, HashSet<Integer>> serverToBuckets =
-        groupByServerToBuckets(prAdvisor, bucketToKeysMap.keySet(), primaryMembersNeeded);
+        groupByServerToBuckets(prAdvisor, bucketToKeysMap.keySet(), primaryMembersNeeded, region);
 
     if (serverToBuckets == null) {
+      logger.debug("One or more primary bucket locations are unknown "
```

Comment:
Add a "if (debug)" check to prevent calling "region.getFullPath()" when debug is not enabled

Reply:
sure - done